### PR TITLE
[saddle-finance-v2] New Snapshot Strategy With Logic Fix

### PR DESCRIPTION
**Pull request by ug02fast**
Changes proposed in this pull request:
- Change logic away from taking into consideration total vest amount to unclaimed and vested amount.
Closes https://github.com/saddle-finance/tickets/issues/244

### CODE REVIEW

**Comment by ChaituVR** [comment]
Hi @ug02fast This can potentially change voting power in existing proposals (including the one that is still active) so better to create a new `-v2` strategy

**Comment by ug02fast** [comment]
> Hi @ug02fast This can potentially change voting power in existing proposals (including the one that is still active) so better to create a new `-v2` strategy

Good idea. Thanks.

File: src/strategies/saddle-finance/index.ts [hunk 1]
```diff
@@ -49,25 +50,24 @@ export async function strategy(
...
 
-  const beneficiaries = multicall(
+  const vestedAndUnclaimedAmountRes = multicall(
     network,
     provider,
     abi,
     vestingContractAddrs.map((vestingContractAddress: any) => [
       vestingContractAddress.toLowerCase(),
-      'beneficiary'
+      'vestedAmount'
```
Comment (hammeiam) [inline src/strategies/saddle-finance/index.ts]: The logic here would need to be to get both the `vestedAmount` as well as `released` from the same contract, and use `walletBalance + vestedAmount - released` as the voting amount

Comment (hammeiam) [reply src/strategies/saddle-finance/index.ts]: I was incorrect about this logic ^

**Review by bonustrack — APPROVED** [review]

**Pull request merged**